### PR TITLE
requirements.txt: add riotctrl as a dependency

### PR DESCRIPTION
**Pull request by miri64**
This adds [`riotctrl`](https://github.com/RIOT-OS/riotctrl/) to the docker image.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -14,3 +14,4 @@ protobuf==3.10.0
 scikit-learn==0.22.1
 joblib==0.14.0
 emlearn==0.10.1
+git+https://github.com/RIOT-OS/riotctrl
```
Comment (aabadie) [inline requirements.txt:17]: Would it makes sense if we first release it on pypi and use this version instead ?

Comment (miri64) [reply requirements.txt:17]: Does it make sense to release it on pypi if most likely we will only use this internally?

Comment (aabadie) [reply requirements.txt:17]: For compatibility between versions, I think so. If the ifconfig output of a given version of RIOT is changed, then previous RIOT version will always fail. Having a specific RIOT version linked to a riotctrl version would prevent that. And we would also have to use this version the Dockerfile and link each RIOT version to a Docker image version. This we are not doing for the moment.

Comment (miri64) [reply requirements.txt:17]: > For compatibility between versions, I think so. If the ifconfig output of a given version of RIOT is changed, then previous RIOT version will always fail. Having a specific RIOT version linked to a riotctrl version would prevent that. And we would also have to use this version the Dockerfile and link each RIOT version to a Docker image version. This we are not doing for the moment.

`riotctrl_shell` is bound to the RIOT version, not the `riotctrl` version. Thus `ifconfig`'s output is bound to RIOT, not `riotctrl`. So I'm not sure what you mean.

Comment (miri64) [reply requirements.txt:17]: We could also link a tag here, if you want to link to a static version.

Comment (aabadie) [reply requirements.txt:17]: > So I'm not sure what you mean

You are right, my assumption was wrong. I think that I had in mind to move all shell interation code from RIOT to riotctrl. But maybe that's useless.

Comment (miri64) [reply requirements.txt:17]: No, that was never the idea and that is not how I remember the discussions about https://github.com/RIOT-OS/RIOT/pull/10949 concluded.

Comment (aabadie) [reply requirements.txt:17]: > We could also link a tag here, if you want to link to a static version.

We could publish on pypi for each new tag of riotctrl :)

Comment (miri64) [reply requirements.txt:17]: (for the very same reason we draw out here: it is better to keep the input/output handling for specific shell commands with the shell commands)

Comment (miri64) [reply requirements.txt:17]: > > We could also link a tag here, if you want to link to a static version.
> 
> We could publish on pypi for each new tag of riotctrl :)

We can, but again: I don't see the point of that for a very RIOT-specific library.

Comment (miri64) [reply requirements.txt]: After some reconsideration [`riotctrl`](https://pypi.org/project/riotctrl/) is now on PyPI (adding it as a dependency in `setup.py` is harder with github links I believe). I changed the line accordingly.

**Comment by kaspar030** [comment]
pls rebase and squash

**Comment by benpicco** [comment]
Please squash (& rebase)

**Comment by miri64** [comment]
Rebased (rebased!) and squashed (_aaand_ squashed)

(and updated `riotctrl` minimal version)

**Pull request merged**